BitwardenShared/Core/Autofill/Services/AutofillCredentialService.swift

```diff
@@ -225,6 +244,85 @@ extension DefaultAutofillCredentialService: AutofillCredentialService {
 
         return ASPasswordCredential(user: username, password: password)
     }
+
+    @available(iOS 17.0, *)
+    func provideFido2Credential( // swiftlint:disable:this function_body_length
+        for passkeyRequest: ASPasskeyCredentialRequest,
+        withUserInteraction: Bool,
+        fido2UserVerificationMediatorDelegate: Fido2UserVerificationMediatorDelegate
+    ) async throws -> ASPasskeyAssertionCredential {
+        guard let credentialIdentiy = passkeyRequest.credentialIdentity as? ASPasskeyCredentialIdentity else {
+            throw AppProcessorError.invalidOperation
+        }
+
+        let isLocked = try? await authRepository.isLocked()
```

Comment:
🎨 Could we use `vaultTimeoutService.isLocked()` here instead of adding the auth repository? I think it's better if we don't have services depend on repositories.

Reply:
🤔 But how would I replace `authRepository.unlockVaultWithNeverlockKey`?